### PR TITLE
Booking Prisma and Server Actions

### PR DESCRIPTION
**Pull request by Luceium**

### CODE REVIEW

**Comment by Luceium** [comment]
```typescript
const dummySearchResults = [...] // populate this with tmp data that should have the same schema as the expected api results (see api docs for this)

// Example usage
const availbleRooms = dummySearchResults.filter(room => roomAvailbleDuringDates(room.id, startDate, endDate))

// Your task:
// This function should be a server action (see Next.js docs)
export function async roomAvailbleDuringDates(id: number, startDate: Date, endDate: Date) {
    ...
    // use prisma methods
    // if there are any bookings in our DB for that room during our requested dates then return false
    // it may be more optimal to only search through bookings for a specific hotel
}
// Note this function can be a lot more optimal if it returns all valid rooms instead of true or false
// the example expecting true or false is just to illustrate the behaviour and necessity.
```

File: roomAvailability.js
Comment (ray-sjsu) [inline roomAvailability.js:1]: Delete this file, we don't need this anymore since `.ts` version of this already exists.

File: roomAvailability.ts
Comment (ray-sjsu) [inline roomAvailability.ts:1]: https://nextjs.org/docs/app/building-your-application/routing/route-handlers

You would have to rename this to `route.ts`. And put in this directory: `app/api/availability`. See the link above to see how api's work in nextjs.

Also, change the `prisma.schema` to this structure. You can find this in `/lib/rapid-hotel-api/api-setup.ts` and import it into this `.ts`. This is what we will use. Update this `.ts` file accordingly to match the schema. Please make sure to fix any typescript errors before any commits.
image

**Review by ray-sjsu — CHANGES_REQUESTED** [review]

File: roomAvailability.ts
Comment (alia49) [reply roomAvailability.ts:1]: I wasn't able to find the api-setup.ts. But is the rest of the script good? it should work if I can find the id's

**Comment by ray-sjsu** [comment]
Currently, this branch is outdated from `main`. Also, rework of reservation system in progress.

https://github.com/KingSand08/LikeHome/pull/98 - new reservation system implementation (basic).

**Pull request closed without merging**